pkg/stanza/docs/operators/file_input.md

```diff
@@ -63,6 +67,13 @@ To avoid the data loss, choose move/create rotation method and set `max_concurre
 
 Other less common encodings are supported on a best-effort basis. See [https://www.iana.org/assignments/character-sets/character-sets.xhtml](https://www.iana.org/assignments/character-sets/character-sets.xhtml) for other encodings available.
 
+### Header Metadata Parsing
+
+To enable header metadata parsing, the `filelog.allowHeaderMetadataParsing` feature gate must be set.
+
+If set, the file input operator will attempt to read a header from the start of the file. Each header line must match the `header.line_start_pattern` pattern. After the header is fully read, the full multiline header is set to the `body` of an entry, and emitted into an embedded pipeline specified by `header.metadata_operators`. Any attributes on the resultant entry from the embedded pipeline will be used as initial attributes for any log lines emitted from this file.
```

Comment:
I think initially, it would have helped with sharing a pipeline between multiple headers. I moved away from that since then, so it could make sense to do a line-by-line thing.

The other reason why the multiline entry is nice is that we don't have to define how we are merging attributes from all the log lines. Not sure which is more confusing to a user, I'm certainly open to the line-by-line approach if we are OK with doing something like just overwriting keys in the resultant map to merge attributes from multiple entries.

Reply:
> just overwriting keys in the resultant map to merge attributes

I think this is ok. We handle merge collisions in other cases with the understanding that it is an upsert operation. Seems reasonable to use the same here.

Reply:
`header.pattern` makes sense to me, or maybe `header.regex`

Reply:
changed to `header.pattern`